lib/matplotlib/backends/backend_pdf.py

```diff
@@ -2301,7 +2302,13 @@ def delta(self, other):
                 if different:
                     break
 
+            # Need to update hatching if we also updated fillcolor
+            if params == ('_hatch',) and fill_performed:
```

Comment:
I believe so. The writeHatches function seems to independently change the fill color (line 1189), so requires updating the hatching if the fill color is modified.